### PR TITLE
optimize memory usage

### PR DESCRIPTION
**Pull request by DFSko**
In structural logging message is decomposed
Final message, original message and each variable will be logged separately
Each of them will allocate memory
So, huge response body in message will be allocated twice

### CODE REVIEW

File: HappyTravel.HttpRequestLogger/HttpRequestLoggingHandler.cs [hunk 1]
```diff
@@ -105,6 +110,24 @@ void WriteLog(HttpRequestLogEntry entry, Exception? exception = null)
...
+        private static string ConvertToString(Dictionary<string, string>? dictionary)
+        {
+            if (dictionary is null)
+                return string.Empty;
+
+            var sb = new StringBuilder();
+
+            foreach (var (key, value) in dictionary)
+            {
+                sb.Append(key);
```
Comment (kirillta) [inline HappyTravel.HttpRequestLogger/HttpRequestLoggingHandler.cs:122]: If I remember right, one may append a template string like `AppendFormat("{0}:{1}{2}", key, value, Environment.NewLine)`. Or without a new line symbol.

File: HappyTravel.HttpRequestLogger/HttpRequestLoggingHandler.cs [hunk 2]
```diff
@@ -105,6 +110,24 @@ void WriteLog(HttpRequestLogEntry entry, Exception? exception = null)
                 h => string.Join(";", h.Value));
         }
 
+        private static string ConvertToString(Dictionary<string, string>? dictionary)
+        {
+            if (dictionary is null)
+                return string.Empty;
+
+            var sb = new StringBuilder();
```
Comment (kirillta) [inline HappyTravel.HttpRequestLogger/HttpRequestLoggingHandler.cs:118]: I wonder if this string builder could be pooled

File: HappyTravel.HttpRequestLogger/HttpRequestLoggingHandler.cs (hunk 1 above)
Comment (DFSko) [reply HappyTravel.HttpRequestLogger/HttpRequestLoggingHandler.cs]: Fixed

File: HappyTravel.HttpRequestLogger/HttpRequestLoggingHandler.cs (hunk 2 above)
Comment (DFSko) [reply HappyTravel.HttpRequestLogger/HttpRequestLoggingHandler.cs]: Fixed

**Review by AlexeyShok — APPROVED** [review]

**Pull request merged**